Makefile

```diff
@@ -28,16 +25,35 @@ test: pre-build-tests post-build-tests
 valid: pre-build-tests-fast build post-build-tests-fast
 
 ## `make all`: build and run all tests
-all: travis-background-keepalive pre-build-tests build post-build-tests
+all: pre-build-tests build post-build-tests
+
+## `make deployment`: for use on build server
+deployment: install-deps-deployment \
+    valid
+
+## `make travis`: for use with Travis CI
+travis: travis-background-keepalive \
+    install-deps-development \
+    all
+
 
 
 
-## Pre-build tests which, aggregated together, take less than 5 seconds to run on a typical PC
+## Install dependencies (development version)
+install-deps-development:
+	bundle install
+
+## Install dependencies (deployment version)
+install-deps-deployment:
+	bundle install --deployment --without :slow_test
+
```

Comment:
@saivann one is "-**deploy**ment" (used by the build server) and the other is "-**develop**ment" (used, somewhat unnecessarily, by Travis CI).  Travis actually checks out the deps on its own, but I wanted a rule that replicated what it does for local testing.